Add support for segno reading

SemitoneGene opened the pull request:
Segno are now read and made available in the public API.

SemitoneGene commented:
I also made a fix to catch.h in cpul that was causing a crash when building on the M1.

SemitoneGene commented:
Coda reading added as well.

SemitoneGene commented:
Added rehearsal reading.

In typical Gene fashion, I just realized all these changes only deal with reading. I am working on getting the writer work done.

SemitoneGene commented:
I stubbed in the writing functions, but need to do some more work to understand the correct way to write the info out. Any help is appreciated!

webern commented:
I'll take a look.

webern commented:
I think I fixed the git history at least. I force pushed to your fork and branch (because you can do that with GitHub, it's crazy).

So, Assuming your fork is at "origin", you should:

```sh
cd mx
git remote add upstream git@github.com:webern/mx.git
git checkout master
git fetch upstream master
git reset --hard upstream/master
git push origin -f master
git fetch origin gene/351
git checkout gene/351
git reset --hard upstream gene/351
```

SemitoneGene committed:
Add segno, coda and rehearsal direction writing

The reading side (DirectionReader, the *Data headers and the
DirectionData vectors) already landed on main, so the rebase reduces
to the writer. Port the per-item write-back for segnos, codas and
rehearsals onto the new core API (DirectionTypeChoice factories taking
OneOrMore<...>). These remain stubs: they emit one empty element per
item and do not yet transfer position, color or text.
